### PR TITLE
Feature/esm button beta

### PR DESCRIPTION
**Pull request by EdwardIrby**
Had to redo this PR to do publish

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -12,7 +13,10 @@ module.exports = api => {
 
     ignore,
 
-    presets: ['@babel/preset-react', '@babel/preset-env'],
+    presets: [
+      '@babel/preset-react',
+      esm ? ['@babel/preset-env', { modules: false }] : '@babel/preset-env'
+    ],
```
Comment (danethurber) [inline babel.config.js]: i would have do it this way

```
presets: [
      '@babel/preset-react',
      ['@babel/preset-env', { modules: !esm }]
    ],
```

your call

File: packages/button/package.json [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
+  "main": "react.js",
+  "module": "index.js",
   "scripts": {
-    "build": "run-s build:js build:css",
-    "build:css": "build-css --useGlamor",
-    "build:js": "babel --root-mode upward src --out-dir dist",
+    "build": "run-s build:js build:cjs build:css",
+    "build:cjs": "babel --root-mode upward src --out-dir dist/cjs",
+    "build:css": "build-css --useGlamor -i dist/cjs/css/index.js",
+    "build:js": "cross-env ESM=true babel --root-mode upward src --out-dir dist/esm",
```
Comment (danethurber) [inline packages/button/package.json]: is the name `build:esm` better? maybe? idk

Comment (EdwardIrby) [reply packages/button/package.json]: I think the idea is to move every component to esm so I decided to change the name of the old build to reflect it's the fallback `build:cjs`

File: babel.config.js (hunk 1 above)
Comment (EdwardIrby) [reply babel.config.js]: modules argument cannot be a boolean value of `true` I'll switch this to a ternary in the object value though.

**Pull request merged**